specification/azureactivedirectory/resource-manager/Microsoft.Aadiam/stable/2017-04-01/azureactivedirectory.json

```diff
@@ -37,6 +37,117 @@
     }
   },
   "paths": {
+    "/providers/microsoft.aadiam/privateLinkForAzureAD": {
```

Comment:
>privateLinkForAzureAD [](start = 33, length = 21)

Are you planning to support non-AzureAD private links? If the answer is no, you already have AAD in the namespace, so this is kind of redundant. On top of that, other RPs are calling this `privateLinks`, so it would be great if private link setup was at least somewhat consistent.

Reply:
@KranthiKiranerusu: This feedback makes sense. If we want to just call it private link, that's fine. @majastrz : Would we still able to give it a "display" name of "Private Link for Azure AD"?

Reply:
Yeah, should be able to. I assume you're talking about the asset display name in the portal that maps to a resource type - that is completely controlled by the UX extension that publishes the asset as far as I know. 

---
In reply to: [403360887](https://github.com/Azure/azure-rest-api-specs/pull/8854#discussion_r403360887) [](ancestors = 403360887)

Reply:
Hmm... So I agree with your suggestion, that said, I just realized that our "privateLinkforAzureAD" name has already been given to NRP for onboarding us to private link and they manage to get us to canary while under lockdown. If we have to change our name now, it will probably delay is a few more weeks.

Is the problem with the name purely about the redundant wording?

Reply:
This is why RP teams should be getting swaggers approved (or at least start the reviews) before any work is done. This way feedback can actually be incorporated.

The good news is that the situation is better than I thought initially. I looked over how other RPs are doing this and there is standardization on /privateLinkResources (which you're doing also), but the parent resource type is always different. I don't particularly like the name, but I think it's less of an issue at this point. The other rule you're violating with `privateLinkForAzureAd` that the name is singular - they're supposed to be plural (like `subscriptions`, `resourceGroups`, or `virtualMachines`), but I guess it'll have to stay like this now...

---
In reply to: [403389682](https://github.com/Azure/azure-rest-api-specs/pull/8854#discussion_r403389682) [](ancestors = 403389682)

Reply:
RE: "This is why RP teams should be getting swaggers approved (or at least start the reviews) before any work is done. This way feedback can actually be incorporated."

Totally valid. That was our fault for not checking in earlier. JEDI + COVID made our wires get crossed on order for this. 

In the future, would we be able to rename it? Might be a bigger problem at that point?

Reply:
Since it's a tenant-level proxy resource, it would be fairly easy to make one underlying resource appear under two different names in ARM. How NRP would deal with this, I have no idea. 

---
In reply to: [403395988](https://github.com/Azure/azure-rest-api-specs/pull/8854#discussion_r403395988) [](ancestors = 403395988)

Reply:
@majastrz , this is totally new to us and we are pressed against deadlines from JEDI and NRP alike. since their onboarding is in progress we woudl like to go ahead with privateLinkForAzureAD.  If an opprotunity presents in the next 4 weeks we will add a new resource with your suggestion and deprecate the old one while in private preivew.